File: Customer Landing Page/patch_trust_pages.py
import sys

file_path = 'src/App.jsx'
with open(file_path, 'r') as f:
    content = f.read()

replacements = [
  ('<PrivacyPolicyPublicPage />', '<PrivacyPolicyPublicPage location={location} onLocationClick={() => setIsLocationModalOpen(true)} />'),
  ('<ServiceWarrantyPolicyPage />', '<ServiceWarrantyPolicyPage location={location} onLocationClick={() => setIsLocationModalOpen(true)} />'),
  ('<VerifiedSellersPage />', '<VerifiedSellersPage location={location} onLocationClick={() => setIsLocationModalOpen(true)} />'),
  ('<SecureOnlinePaymentsPage />', '<SecureOnlinePaymentsPage location={location} onLocationClick={() => setIsLocationModalOpen(true)} />'),
  ('<PrivacyProtectedPage />', '<PrivacyProtectedPage location={location} onLocationClick={() => setIsLocationModalOpen(true)} />'),
  ('<CustomerSupportPage />', '<CustomerSupportPage location={location} onLocationClick={() => setIsLocationModalOpen(true)} />'),
]

for old, new in replacements:
    content = content.replace(old, new)

with open(file_path, 'w') as f:
    f.write(content)

File: Customer Landing Page/patch_trust_pages.js
const fs = require('fs');
const file = 'src/App.jsx';
let content = fs.readFileSync(file, 'utf8');

const replacements = [
  ['<PrivacyPolicyPublicPage />', '<PrivacyPolicyPublicPage location={location} onLocationClick={() => setIsLocationModalOpen(true)} />'],
  ['<ServiceWarrantyPolicyPage />', '<ServiceWarrantyPolicyPage location={location} onLocationClick={() => setIsLocationModalOpen(true)} />'],
  ['<VerifiedSellersPage />', '<VerifiedSellersPage location={location} onLocationClick={() => setIsLocationModalOpen(true)} />'],
  ['<SecureOnlinePaymentsPage />', '<SecureOnlinePaymentsPage location={location} onLocationClick={() => setIsLocationModalOpen(true)} />'],
  ['<PrivacyProtectedPage />', '<PrivacyProtectedPage location={location} onLocationClick={() => setIsLocationModalOpen(true)} />'],
  ['<CustomerSupportPage />', '<CustomerSupportPage location={location} onLocationClick={() => setIsLocationModalOpen(true)} />'],
];

replacements.forEach(([from, to]) => {
  content = content.split(from).join(to);
});

fs.writeFileSync(file, content);
